### PR TITLE
Fix coverity warnings caused by intentional missing breaks

### PR DESCRIPTION
**Pull request by stweil**
Signed-off-by: Stefan Weil <sw@weilnetz.de>

### CODE REVIEW

**Comment by henning-gerhardt** [comment]
@matthias-ronge : Can you please review this pull request?

File: Goobi/src/de/sub/goobi/forms/GranularityForm.java [hunk 1]
```diff
@@ -199,12 +199,16 @@ public void downloadClick() {
...
+				// fall through
 			case DAYS:
 				result.add(new SelectItem("weeks", Helper.getTranslation("granularity.weeks")));
+				// fall through
 			case WEEKS:
 				result.add(new SelectItem("months", Helper.getTranslation("granularity.months")));
+				// fall through
 			case MONTHS:
 				result.add(new SelectItem("quarters", Helper.getTranslation("granularity.quarters")));
+				// fall through
```
Comment (matthias-ronge) [inline Goobi/src/de/sub/goobi/forms/GranularityForm.java]: All fall throughs are correct.

File: Goobi/src/de/sub/goobi/metadaten/copier/MetadataPathSelector.java [hunk 2]
```diff
@@ -349,26 +349,21 @@ private DocStruct getSubnode(DocStruct logicalNode) {
...
-			}
 		}
+
+		if (index == null) {
+			throw new RuntimeException("Could not resolve metadata path: Path selector is ambiguous for "
+					+ docStructType);
+		} else if (!(index instanceof Integer)) {
+			throw new RuntimeException("Could not resolve metadata path: In this regard, index \""
+					+ index + "\" is not allowed.");
+		} else if (index.equals(Long.MAX_VALUE)) {
```
Comment (matthias-ronge) [inline Goobi/src/de/sub/goobi/metadaten/copier/MetadataPathSelector.java]: Rereading the code, I found a bug here: this must be `index.equals(Integer.MAX_VALUE)`—cf. other uses in this file. May you correct this in the commit?

File: Goobi/src/de/sub/goobi/metadaten/copier/MetadataPathSelector.java [hunk 3]
```diff
@@ -349,26 +349,21 @@ private DocStruct getSubnode(DocStruct logicalNode) {
...
+			return children.get(children.size() - 1);
+		} else if (children.size() >= ((Integer) index).intValue()) {
+			return children.get(((Integer) index).intValue());
+		}
+
+		return null;
```
Comment (matthias-ronge) [inline Goobi/src/de/sub/goobi/metadaten/copier/MetadataPathSelector.java]: I think this removal of braces is semantically equivalent.

File: Goobi/src/de/sub/goobi/metadaten/copier/MetadataPathSelector.java (hunk 2 above)
Comment (stweil) [inline Goobi/src/de/sub/goobi/metadaten/copier/MetadataPathSelector.java]: This is a separate issue, so I'd prefer to handle it in a separate commit.
@henning-gerhardt, you can either pull this PR now or wait until I add that separate commit in the afternoon.

Comment (henning-gerhardt) [inline Goobi/src/de/sub/goobi/metadaten/copier/MetadataPathSelector.java]: I will merge this and the mentioned bug should be fixed by different pull request.

Comment (stweil) [inline Goobi/src/de/sub/goobi/metadaten/copier/MetadataPathSelector.java]: The bug fix is handled by PR #546.

**Pull request merged**